pkg/build/webhook/generic/generic.go

```diff
@@ -44,7 +45,10 @@ func (p *WebHookPlugin) Extract(buildCfg *api.BuildConfig, secret, path string,
 		return nil, true, nil
 	}
 
-	if req.Body != nil && req.Header.Get("Content-Type") == "application/json" {
+	contentType := req.Header.Get("Content-Type")
+	mediaType, _, _ := mime.ParseMediaType(contentType)
```

Comment:
Generic webhook was always more forgiving - if it wasn't exactly `application/json` then it still fires, just with less info. Guess could add the err check into the following conditional if you'd prefer than just relying on `mediaType` being empty on failure?